src/server/req_manager.c

```diff
@@ -2876,6 +2878,41 @@ make_host_addresses_list(char *phost, u_long **pul)
 		return 0;
 }
 
+/**
+ * @brief
+ * 		remove the cached ip addresses of a mom from the host tree and the ipaddrs tree
+ *
+ * @param[in]	pmom - ptr to the mom info
+ *
+ * @return	error code
+ * @retval	0			- no error
+ * @retval	PBS error	- error
+ */
+int
+remove_mom_ipaddresses_list(mominfo_t *pmom)
+{
+	/* take ipaddrs from ipaddrs cache tree */
+	if (hostaddr_tree != NULL) {
+		struct pul_store *tpul;
+		u_long *pul;
+
+		if ((tpul = (struct pul_store *) find_tree(hostaddr_tree, pmom->mi_host)) != NULL) {
+			pul = tpul->pul;
+			while (*pul) {
```

Comment:
You're iterating.  Use a for() loop.

Reply:
okay - i know you love for() way more... :)

Reply:
And I know you're teasing a bit, but I've promoted the same thing several times in reviews. All it takes is someone to throw a "continue" in a while loop, and poof... infinite loop. The for loop protects against that.

Reply:
of course and i like for too :)